Lib/test/test_inspect.py

```diff
@@ -4157,16 +4150,9 @@ def test(a, *args, b, z=100, **kwargs):
     def test_signature_bind_positional_only(self):
         P = inspect.Parameter
 
-        def test(a_po, b_po, c_po=3, foo=42, *, bar=50, **kwargs):
+        def test(a_po, b_po, c_po=3, /, foo=42, *, bar=50, **kwargs):
             return a_po, b_po, c_po, foo, bar, kwargs
 
-        sig = inspect.signature(test)
```

Comment:
I think it's worth adding a new test that's explicitly designed to check that if you modify `__signature__`, `inspect.signature` retrieves the modified signature. I don't think there's other tests in `test_inspect` that directly test that behavior.

Reply:
I think that this test already does this: https://github.com/python/cpython/blob/ecad802e3f9ef7063e7d7fe9f0c037561396ef8e/Lib/test/test_inspect.py#L1045-L1051 :)

I've mentioned it in my PR description 👍

Reply:
I saw that but I feel it's testing `getfullargspec`, not `signature`.

Reply:
Ok, new tests are always a good thing, adding :)

Reply:
Done!